### PR TITLE
Add doc about extension backend mounting docker socket

### PR DESCRIPTION
**Pull request by gtardif**
Signed-off-by: Guillaume Tardif <guillaume.tardif@gmail.com>

### CODE REVIEW

File: docs/extensions/METADATA.md [hunk 1]
```diff
@@ -80,7 +88,30 @@ For more advanced use cases, the extension can also:
...
+services:
+  myExtension:
+    image: ${DESKTOP_PLUGIN_IMAGE}
+    volumes:
+      - /host/path:/container/path
+```
+
+### Using the docker socket from your extension backend
+
+Docker extensions can invoke Docker commands directly from the UI js API. In some cases it can be useful to also interact with the Docker engine from the backend. Extension backend containers can mount the docker socket and use it to interact with the Docker engine from the extension backend logic.
```
Comment (benja-M-1) [inline docs/extensions/METADATA.md:103]:
```suggestion
Docker extensions can invoke Docker commands directly from the frontend with the SDK. In some cases, it is useful to also interact with the Docker engine from the backend. Extension backend containers can mount the docker socket and use it to interact with the Docker engine from the extension backend logic.
```

Comment (benja-M-1) [inline docs/extensions/METADATA.md:103]: Shouldn't we explain what the docker socket is for developers that don't know how docker engine and docker desktop internals work?

**Review by benja-M-1 — COMMENTED** [review]
The explanation about `docker.sock` assumes the reader knows how docker actually works. Is there a doc on the architecture we can refer to so that they can understand the big picture?

File: docs/extensions/METADATA.md (hunk 1 above)
Comment (gtardif) [reply docs/extensions/METADATA.md:103]: I would assume developers who don't know what it is will likely not need this level of information, and need to learn more about Docker in general in order to develop complex extensions. (or they would just execute the docker CLI from the SDK for simple use cases)
I'm not sure if we should point them at some general Docker tutorial... I would assume if you read this you already know about the general Docker mechanisms

**Comment by gtardif** [comment]
> The explanation about `docker.sock` assumes the reader knows how docker actually works. Is there a doc on the architecture we can refer to so that they can understand the big picture?
Added a "learn more" link to https://docs.docker.com/engine/reference/commandline/dockerd/#examples

File: docs/extensions/METADATA.md [hunk 2]
```diff
@@ -65,8 +69,12 @@ If you use the same image for the extension packaging and for the backend servic
 },
 ```
 
+The vm metadata section should define either `image` or `composefile`. When you use `image`, a default compose file is generated for the extension.
+
 > `${DESKTOP_PLUGIN_IMAGE}` is a specific keyword that allows an easy way to refer to the image packaging the extension. It is also possible to specify any other full image name here. However, in many cases using the same image makes things easier for extension development.
 
+### Defining your own compose file for extension backend
```
Comment (aevesdocker) [inline docs/extensions/METADATA.md:76]:
```suggestion
### Define your own compose file for the extension backend
```

File: docs/extensions/METADATA.md [hunk 3]
```diff
@@ -80,7 +88,30 @@ For more advanced use cases, the extension can also:
 },
 ```
 
-The vm metadata section should define either `image` or `composefile`. When you use `image`, a default compose file is generated for the extension.
+The composefile would then look like (for exampe with a volume definition):
```
Comment (aevesdocker) [inline docs/extensions/METADATA.md:91]:
```suggestion
The composefile, with a volume definition for example, would look like:
```

File: docs/extensions/METADATA.md [hunk 4]
```diff
@@ -80,7 +88,30 @@ For more advanced use cases, the extension can also:
...
+
+```yaml
+services:
+  myExtension:
+    image: ${DESKTOP_PLUGIN_IMAGE}
+    volumes:
+      - /host/path:/container/path
+```
+
+### Using the docker socket from your extension backend
```
Comment (aevesdocker) [inline docs/extensions/METADATA.md:101]:
```suggestion
### Use the docker socket from your extension backend
```

File: docs/extensions/METADATA.md [hunk 5]
```diff
@@ -80,7 +88,30 @@ For more advanced use cases, the extension can also:
...
+
+Docker extensions can invoke Docker commands directly from the frontend with the SDK. In some cases, it is useful to also interact with the Docker engine from the backend. Extension backend containers can mount the Docker socket and use it to interact with the Docker engine from the extension backend logic. (Learn more about the [Docker engine socket](https://docs.docker.com/engine/reference/commandline/dockerd/#examples))
+
+However, when mounting the docker socket from an extension container that lives in the Desktop virtual machine, you want to mount the Docker socket from inside the VM, and not mount `/var/run/docker.sock` from the host filsystem (using the Docker socket from the host can lead to permission issues in containers).
+In order to do so, you can use `/var/run/docker.sock.raw` and Docker Desktop will treat this specifically to mount the socket that lives in the Desktop VM, and not from the host.
```
Comment (aevesdocker) [inline docs/extensions/METADATA.md:106]: Not sure about this sentence. I've added a suggestion, but not sure if this then changes the meaning of the sentence?
```suggestion
In order to do so, you can use `/var/run/docker.sock.raw`. Docker Desktop mounts the socket that lives in the Desktop VM, and not from the host.
```

**Review by aevesdocker — APPROVED** [review]
LGTM

**Pull request merged**